index.js

```diff
@@ -243,6 +246,49 @@ module.exports = function(version, _options) {
             }
 
             return output;
+        },
+        getBestMatchingLanguage: function(language) {
+            if (languages.instructions[language]) return language;
+
+            var codes = languages.parseLanguageIntoCodes(language);
+            var languageCode = codes.languageCode;
+            var scriptCode = codes.scriptCode;
+            var countryCode = codes.countryCode;
+
+            var supportedLanguageCodes = languages.supportedCodes.map(function(language) {
+                return language.toLowerCase().split('-')[0];
+            });
+
+            // Same language code and script code (lng-Scpt)
+            if (languages.instructions[languageCode + '-' + scriptCode]) {
+                return languageCode + '-' + scriptCode;
+            // Same language code and country code (lng-CC)
+            } else if (languages.instructions[languageCode + '-' + countryCode]) {
+                return languageCode + '-' + countryCode;
+            // Same language code (lng)
+            } else if (supportedLanguageCodes[languageCode]) {
+                return languageCode;
+            // Same language code and any script code (lng-Scpx) and the found language contains a script
+            } else if (languages.parsedSupportedCodes.find(function (language) {
+                return language.language === languageCode && language.scriptCode;
+            })) {
+                return languages.parsedSupportedCodes.find(function (language) {
+                    return language.languageCode === languageCode && language.scriptCode;
+                }).locale;
```

Comment:
Instead of repeating the `find()` call, store the result of the first find. It’d be a lot easier to store things in local variables if you replace all the `else if`s with `if`s – after all, we’re returning early in each case.